s3/src/test/scala/akka/stream/alpakka/s3/scaladsl/S3IntegrationSpec.scala

```diff
@@ -692,6 +696,112 @@ trait S3IntegrationSpec
       fullInputData.utf8String shouldEqual fullDownloadedFile.utf8String
     }
   }
+  @tailrec
+  final def createStringCollectionContextWithMinChunkSizeRec(
```

Comment:
We could probably abstract the core algorithm for creating the chunks along with `createStringCollectionWithMinChunkSizeRec`to reduce boilerplate but I wanted to get this PR out fast so we don't delay release.